atomicapp/nulecule/container.py

```diff
@@ -90,7 +90,7 @@ def pull(self, image, update=False):
 
         cockpit_logger.info('Skipping pulling docker image: %s' % image)
 
-    def extract(self, image, source, dest, update=False):
+    def extract(self, image, source, dest, update=False, file=False):
```

Comment:
ok after some thought I think we need to refactor this function. unfortunately this isn't a generic function for copying out data from a container. It appears that way but we have all of the `App exists locally. Performing update...` stuff in the middle. This should be broken out into two functions. 

- One. This function just copies a file or directory out of a container to a target file or directory (the specified directory could be a tmpdir that will be later cleaned up by the calling function). We may even choose to put this function in a different file as it may belong somewhere else.
- Two. This function creates a tmp dir, calls the first function, does the `App exists locally. Performing update...` and then cleans up the tmp dir. 

Does that make sense?